std/random.d

```diff
@@ -1361,17 +1620,126 @@ how excellent the source of entropy is.
     }
     else
     {
-        import core.thread : Thread, getpid, MonoTime;
-        static bool seeded;
-        static MinstdRand0 rand;
-        if (!seeded)
+        return cast(uint) .unpredictableSeed!ulong;
+    }
+}
+/// ditto
+template unpredictableSeed(UIntType)
+if (isUnsigned!UIntType)
+{
+    static if (!is(Unqual!UIntType == UIntType))
+        alias unpredictableSeed = .unpredictableSeed!(Unqual!UIntType);
+    else static if (is(UIntType == uint))
+        alias unpredictableSeed = .unpredictableSeed;
+    else
+    {
+        /// ditto
+        @property UIntType unpredictableSeed() @nogc nothrow @trusted
         {
-            uint threadID = cast(uint) cast(void*) Thread.getThis();
-            rand.seed((getpid() + threadID) ^ cast(uint) MonoTime.currTime.ticks);
-            seeded = true;
+            version (AnyARC4Random)
+            {
+                static if (UIntType.sizeof <= uint.sizeof)
+                {
+                    return cast(UIntType) arc4random();
+                }
+                else
+                {
+                    UIntType result = void;
+                    arc4random_buf(&result, UIntType.sizeof);
+                    return result;
+                }
+            }
+            else static if (!is(UIntType == ulong))
+            {
+                return cast(UIntType) unpredictableSeed!ulong;
+            }
+            else
```

Comment:
^^^ Just to ping, either this question has been overlooked, or I missed the answer (in which case, sorry!).  Is there a reason to prefer using the system entropy only to seed splitmix, rather than directly as the source of the unpredictable seed?

Reply:
I think the full answer is that it varies by platform. Defaulting to SplitMix will always work reasonably well, so it's a safe generic implementation, but based on benchmarks we might want to add special cases on specific platforms to use the system entropy source directly instead of just using it to seed SplitMix. This is basically what we're already doing with arc4random.